src/java.base/share/classes/java/io/Reader.java

```diff
@@ -140,6 +140,124 @@ public void close() {
         };
     }
 
+    /**
+     * Returns a new {@code Reader} whose source is a {@link CharSequence}.
+     *
+     * <p> The resulting reader is not safe for use by multiple
+     * concurrent threads. If the reader is to be used by more than one
+     * thread it should be controlled by appropriate synchronization.
+     *
+     * <p> The returned reader is initially open. The reader is closed by
+     * calling the {@code close()} method. Subsequent calls to {@code close()}
+     * have no effect.
+     *
+     * <p> After the reader has been closed, the {@code read()},
+     * {@code read(char[])}, {@code read(char[], int, int)},
+     * {@code read(CharBuffer)}, {@code ready()}, {@code skip(long)}, and
+     * {@code transferTo()} methods all throw {@code IOException}.
+     *
+     * <p> The {@code markSupported()} method returns {@code true}.
+     *
+     * @param source {@code CharSequence} providing the character stream.
+     *
+     * @return a {@code Reader} which reads characters from {@code source}
+     *
+     * @throws NullPointerException if {@code source} is {@code null}
+     *
+     * @since 24
+     */
+    public static Reader of(CharSequence source) {
+        Objects.requireNonNull(source);
+
+        return new Reader() {
```

Comment:
Maybe make this into a package‑private `CharSequenceReader` so that [`StringReader​::close()`] doesn’t have to use a try‑catch.

[`StringReader​::close()`]: https://docs.oracle.com/en/java/javase/23/docs/api/java.base/java/io/StringReader.html#close()

Reply:
I don't think that try-catch is too much of a bloat compared to the extra license header and imports from the explicit `CharSequenceReader.java` file.